internal/db/collection_index.go

```diff
@@ -692,3 +692,27 @@ func (db *DB) listAllEncryptedIndexDescriptions(
 
 	return indexes, nil
 }
+
+// reindexNewActiveVersion reindexes all documents in the collection for the new active version.
+func (db *DB) reindexNewActiveVersion(ctx context.Context, col client.CollectionVersion) error {
```

Comment:
todo: This is only needed if there is a migration, which, as far as I know, is really quite rare.

It should also be fairly easy to check.

Please avoid re-indexing if there are no migrations on either this version, or the previously active version - this could save a lot of user/cpu time.

Consider documenting that this check will need to be expanded if/when we allow fields to be mutated.